crates/core/app/tests/mock_consensus.rs

```diff
@@ -36,3 +37,35 @@ async fn mock_consensus_can_send_an_init_chain_request() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn mock_consensus_can_send_a_single_empty_block() -> anyhow::Result<()> {
+    // Install a test logger, and acquire some temporary storage.
+    let guard = common::set_tracing_subscriber();
+    let storage = TempStorage::new().await?;
+
+    // Instantiate the consensus service, and start the test node.
+    let mut engine = {
+        use penumbra_mock_consensus::TestNode;
+        let app_state = AppState::default();
+        let consensus = Consensus::new(storage.as_ref().clone());
+        TestNode::builder()
+            .single_validator()
+            .with_penumbra_auto_app_state(app_state)?
+            .init_chain(consensus)
+            .await?
+    };
```

Comment:
another note: i found that this code is difficult to share between the tests.

this is an ergonomic wrinkle to try ironing out later, but the `C` generic is hard to name. we can probably define an alias in the `common/mod.rs` testing helpers later.

Reply:
hmm, why do we need to name `C`?

couldn't we do something like:
```
let mut engine = TestNode::builder()
            .single_validator()
            .with_penumbra_auto_app_state(AppState::default())?
            .init_chain(Consensus::new(storage.as_ref().clone()))
            .await?
```
and have `C` be inferred from the argument to `init_chain`?

Reply:
we don't _have_ to, either of our examples here work. what i mean to say is that i found `TestNode` to be hard to return from a function. we don't need to do that now, this is just idle thoughts on ergonomics that might come up as we bake this into the `penumbra-app` test suite more.

this isn't pressing, so i'll mark this thread as resolved. :white_check_mark:

Reply:
yep, makes sense, was just curious how it came up